src/client/actions/PresenceUpdate.js

```diff
@@ -0,0 +1,40 @@
+const Action = require('./Action');
+const { Events } = require('../../util/Constants');
+const User = require('../../structures/User');
+
+class PresenceUpdateAction extends Action {
+  handle(data) {
+    let cached = this.client.users.get(data.user.id);
+    if (!cached && data.user.username) cached = this.client.users.add(data.user);
+    if (!cached) return;
+
+    if (data.user && data.user.username) {
+      const other = new User(this.client, data.user);
```

Comment:
I don't think you need to instantiate a new user, equals seems to not care about the prototype.
https://github.com/discordjs/discord.js/blob/01476de58250761ffe5eedc9ee6c9782576ca043/src/structures/User.js#L219-L227